web/app/js/components/ResourceDetail.jsx

```diff
@@ -53,67 +48,146 @@ export class ResourceDetailBase extends React.Component {
+
+    let { resource } = this.state;
+
+    this.api.setCurrentRequests([
+      // inbound stats for this resource
+      this.api.fetchMetrics(
+        `${this.api.urlsForResource(resource.type, resource.namespace)}&resource_name=${resource.name}`
+      ),
+      // list of all pods in this namespace (hack since we can't currently query for all pods in a resource)
+      this.api.fetchPods(resource.namespace),
+      // metrics for all pods in this namespace (hack, continued)
+      this.api.fetchMetrics(
+        `${this.api.urlsForResource("pod", resource.namespace)}`
+      ),
+    ]);
+
+    Promise.all(this.api.getCurrentPromises())
+      .then(([resourceRsp, podListRsp, podRsp]) => {
+        let resourceMetrics = processSingleResourceRollup(resourceRsp);
+        let podMetrics = processSingleResourceRollup(podRsp);
+
+        // INEFFICIENT: get metrics for all the pods belonging to this resource.
+        // Do this by querying for metrics for all pods in this namespace and then filtering
+        // out those pods whose owner is not this resource
+        // TODO: fix (#1467)
+        let podBelongsToResource = _.reduce(podListRsp.pods, (mem, pod) => {
+          if (_.get(pod, resource.type) === resource.namespace + "/" + resource.name) {
+            mem[pod.name] = true;
+          }
+          return mem;
+        }, {});
+
+        let podMetricsForResource = _.filter(podMetrics, pod => podBelongsToResource[pod.namespace + "/" + pod.name]);
+
+        this.setState({
+          resourceMetrics,
+          podMetrics: podMetricsForResource,
+          loaded: true,
+          pendingRequests: false,
+          error: null
+        });
+      })
+      .catch(this.handleApiError);
+  }
+
+  handleApiError = e => {
+    if (e.isCanceled) {
       return;
     }
 
-    return <ErrorBanner message={error} />;
+    this.setState({
+      loaded: true,
+      pendingRequests: false,
+      error: e
+    });
   }
 
-  content = () => {
-    const {data, loading, error} = this.props;
-
-    if (loading && !error) {
-      return <Spin size="large" />;
+  banner = () => {
+    if (!this.state.error) {
+      return;
     }
 
-    let processedMetrics = [];
-    if (_.has(data, '[0].ok')) {
-      processedMetrics = processSingleResourceRollup(data[0]);
+    return <ErrorBanner message={this.state.error} />;
+  }
+
+  content = () => {
+    if (!this.state.loaded && !this.state.error) {
+      return <Spin size="large" />;
     }
 
     return (
-      <MetricsTable
-        resource={this.state.resourceType}
-        metrics={processedMetrics} />
+      <div>
+        <div className="page-section">
+          <MetricsTable
+            resource={this.state.resource.type}
+            metrics={this.state.resourceMetrics} />
+        </div>
+
+        <div className="page-section">
+          <h2 className="subsection-header">Pods</h2>
+          <MetricsTable
+            resource="pod"
+            metrics={this.state.podMetrics} />
```

Comment:
I think we should consider omitting this section if `this.state.resource.type === "pod"`. Otherwise it renders and empty table:

![image](https://user-images.githubusercontent.com/9226/44373798-e98eb800-a49f-11e8-977d-fdaf2bb16d79.png)

Reply:
great point!